pwiz_tools/Skyline/Model/Proteome/ProteinMetadataManager.cs

```diff
@@ -237,7 +247,20 @@ private SrmDocument LookupProteinMetadata(SrmDocument docOrig, IProgressMonitor
                             foreach (var result in FastaImporter.DoWebserviceLookup(proteinsToSearch, progressMonitor, false)) // Resolve them all, now
                             {
                                 Assume.IsTrue(!result.GetProteinMetadata().NeedsSearch());
-                                _processedNodes.Add(docNodesWithUnresolvedProteinMetadata[result].Id.GlobalIndex, result.GetProteinMetadata());
+                                var node = docNodesWithUnresolvedProteinMetadata[result];
+                                var resultMetadata = result.GetProteinMetadata();
+                                if (node.ProteinMetadata is ProteinGroupMetadata proteinGroupMetadata)

```

Comment:
Not so easy to see how this can be written better, but wish it didn't require so much "is" and "as" type checking and casting.